test/end-to-end/core.sh

```diff
@@ -84,193 +85,194 @@ wait_for_registry
+os::cmd::expect_success_and_text "echo '${out}'" 'Using in-cluster configuration'
+os::cmd::expect_success_and_text "echo '${out}'" 'In project test'
+os::cmd::expect_success 'oc delete pod cli-with-token'
+
 out=$(oc run cli-with-token-2 --attach --env=POD_NAMESPACE=test --image=openshift/origin:${TAG} --restart=Never -- cli whoami --loglevel=4 2>&1)
-echo "${out}"
-[ "$(echo $out | grep 'system:serviceaccount:test:default')" ]
-oc delete pod cli-with-token-2
+os::cmd::expect_success_and_text "echo '${out}'" 'system:serviceaccount:test:default'
+os::cmd::expect_success 'oc delete pod cli-with-token-2'
+
 out=$(oc run kubectl-with-token --attach --env=POD_NAMESPACE=test --image=openshift/origin:${TAG} --restart=Never --command -- kubectl get pods --loglevel=4 2>&1)
-echo "${out}"
-[ "$(echo $out | grep 'Using in-cluster configuration')" ]
-[ "$(echo $out | grep 'kubectl-with-token')" ]
+os::cmd::expect_success_and_text "echo '${out}'" 'Using in-cluster configuration'
+os::cmd::expect_success_and_text "echo '${out}'" 'kubectl-with-token'
 
 echo "[INFO] Streaming the logs from a deployment twice..."
-oc create -f test/fixtures/failing-dc.yaml
-tryuntil oc get rc/failing-dc-1
-oc logs -f dc/failing-dc
-wait_for_command "oc get rc/failing-dc-1 --template={{.metadata.annotations}} | grep openshift.io/deployment.phase:Failed" $((60*TIME_SEC))
-oc logs dc/failing-dc | grep 'test pre hook executed'
-oc deploy failing-dc --latest
-oc logs --version=1 dc/failing-dc
+os::cmd::expect_success 'oc create -f test/fixtures/failing-dc.yaml'
+os::cmd::expect_success 'tryuntil oc get rc/failing-dc-1'
+os::cmd::expect_success 'oc logs -f dc/failing-dc'
+os::cmd::try_until_text 'oc get rc/failing-dc-1 --template={{.metadata.annotations}}' 'openshift.io/deployment.phase:Failed'
+os::cmd::expect_success_and_text 'oc logs dc/failing-dc' 'test pre hook executed'
+os::cmd::expect_success 'oc deploy failing-dc --latest'
+os::cmd::expect_success 'oc logs --version=1 dc/failing-dc'
 
 echo "[INFO] Applying STI application config"
-oc create -f "${STI_CONFIG_FILE}"
+os::cmd::expect_success "oc create -f ${STI_CONFIG_FILE}"
 
 # Wait for build which should have triggered automatically
 echo "[INFO] Starting build from ${STI_CONFIG_FILE} and streaming its logs..."
-#oc start-build -n test ruby-sample-build --follow
-os::build:wait_for_start "test"
+os::cmd::try_until_text 'oc get -n test builds' 'Running'
+ 
 # Ensure that the build pod doesn't allow exec
-[ "$(oc rsh ${BUILD_ID}-build 2>&1 | grep 'forbidden')" ]
-os::build:wait_for_end "test"
+BUILD_ID=$(oc get -n test builds  --output-version=v1 --template="{{with index .items 0}}{{.metadata.name}}{{end}}")
+os::cmd::expect_failure_and_text "oc rsh ${BUILD_ID}-build" 'forbidden'
+os::cmd::try_until_text 'oc get -n test builds' 'Complete' $((10*TIME_MIN))
+os::cmd::expect_success "oc build-logs -n test ${BUILD_ID} > ${LOG_DIR}/testbuild.log" # TODO(skuznets): previously we were agnostic about this failing, why?
```

Comment:
@deads2k this was you addition [here](https://github.com/openshift/origin/commit/e781d5e8f0a6cf44a2c80f307b9f381828c18217#diff-b83ac5455d252fc0916d54924166ffa5R523). Thoughts?